mobile/src/board/components/BoardView.tsx

```diff
@@ -1,14 +1,69 @@
 import { Board, GameState } from "../BoardTypes";
-import type { BoardRepository } from "../BoardRepository";
+
+import { Direction } from "../BoardTypes";
 
 import PlayerPawnView from "./PlayerPawnView";
 import TileView from "./TileView";
 
 import "./BoardView.css";
 
+type Listener = (() => Promise<void>) | undefined;
+type ListenerFactory = (line: number, row: number) => Listener;
+
+type RotateRemainingTypeHandler = () => Promise<void>;
+type InsertTileHandler = (direction: Direction, index: number) => Promise<void>;
+
+function createTileListenerFactory(
```

Comment:
This looks like logic that could be embedded inside the `TileView` directly if the `TileView` accepted `line`, `row` and `onInsertTile` props